common/lib/xmodule/xmodule/annotatable_module.py

```diff
@@ -47,29 +71,41 @@ class AnnotatableFields(object):
         default=_('Annotation'),
     )
 
+    uses_xmodule_styles_setup = True
 
-class AnnotatableModule(AnnotatableFields, XModule):
-    js = {
+    preview_view_js = {
         'js': [
             resource_string(__name__, 'js/src/html/display.js'),
             resource_string(__name__, 'js/src/annotatable/display.js'),
             resource_string(__name__, 'js/src/javascript_loader.js'),
             resource_string(__name__, 'js/src/collapsible.js'),
-        ]
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+    preview_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/annotatable/display.scss'),
+        ],
     }
-    js_module_name = "Annotatable"
-    css = {'scss': [resource_string(__name__, 'css/annotatable/display.scss')]}
-    icon_class = 'annotatable'
 
-    def __init__(self, *args, **kwargs):
-        super(AnnotatableModule, self).__init__(*args, **kwargs)
+    studio_view_js = {
+        'js': [
+            resource_string(__name__, 'js/src/raw/edit/xml.js'),
+        ],
+        'xmodule_js': resource_string(__name__, 'js/src/xmodule.js'),
+    }
+    studio_view_css = {
+        'scss': [
+            resource_string(__name__, 'css/codemirror/codemirror.scss'),
+        ],
+    }
+    studio_js_module_name = "XMLEditingDescriptor"
+    mako_template = "widgets/raw-edit.html"
 
-        xmltree = etree.fromstring(self.data)
+    icon_class = 'annotatable'
+    resources_dir = None
 
-        self.instructions = self._extract_instructions(xmltree)
-        self.content = etree.tostring(xmltree, encoding='unicode')
-        self.element_id = self.location.html_id()
-        self.highlight_colors = ['yellow', 'orange', 'purple', 'blue', 'green']
```

Comment:
Since XBlocks should not have an `init`, moved all lines where they were being used.